src/storage/context/StorageExpressionContext.cpp

```diff
@@ -65,9 +66,11 @@ Value StorageExpressionContext::getEdgeProp(const std::string& edgeName,
             return Value::kNullValue;
         }
         if (prop == kSrc) {
-            return NebulaKeyUtils::getSrcId(vIdLen_, key_);
+            auto srcId = NebulaKeyUtils::getSrcId(vIdLen_, key_);
+            return isIntId_ ? srcId : srcId.subpiece(0, srcId.find_first_of('\0'));
```

Comment:
user can insert vid such as "aa\0bb". so graphd need to check vid if has `\0` and return an error to the user.

Reply:
Yeah, graph need to prevent '\0' for string id.